code/obj/machinery/turret.dm

```diff
@@ -467,23 +475,23 @@
 
 
 /obj/machinery/turretid/proc/updateTurrets()
-	if(turretsExist) //Let's not waste a lot of time here.
-		if (src.enabled)
-			if (src.lethal)
-				icon_state = "ai1"
-			else
-				icon_state = "ai3"
-		else
-			icon_state = "ai0"
+	if (!by_type[/obj/machinery/turret].len)
```

Comment:
```suggestion
	if (!length(by_type[/obj/machinery/turret]))
```
length is marginally faster but more importantly won't runtime if by_type[foo] is null

Reply:
Also the line is arguably unnecessary because the for loop handles it already (it will iterate over no items if the list is empty or null).

Reply:
I've scrapped the length check